src/jit/codegencommon.cpp

```diff
@@ -4005,18 +4016,34 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #if FEATURE_MULTIREG_ARGS
             if (varDsc->lvIsMultiregStruct())
             {
+#ifdef FEATURE_HFA
+                if (varDsc->lvIsHfaRegArg)
+                {
+                    // We have an HFA argument, set slots to the number of registers used
+                    slots = varDsc->lvHfaSlots();
+                }
+                else
+#endif // FEATURE_HFA
+                {
+                    // We have a non-HFA multireg argument, set slots to two
+                    slots = 2;
+                }
+
                 // Note that regArgNum+1 represents an argument index not an actual argument register.  
                 // see genMapRegArgNumToRegNum(unsigned argNum, var_types type)
 
-                // This is the setup for the second half of a MULTIREG struct arg
-                noway_assert(regArgNum+1 < regState->rsCalleeRegArgNum);
-                // we better not have added it already (there better not be multiple vars representing this argument register)
-                noway_assert(regArgTab[regArgNum+1].slot == 0);
-                    
-                regArgTab[regArgNum+1].varNum = varNum;
-                regArgTab[regArgNum+1].slot = 2;
 
-                slots = 2;
+                // This is the setup for the rest of a multireg struct arg
+                noway_assert((regArgNum + (slots - 1)) < regState->rsCalleeRegArgNum);
+
+                for (char i = 1; (i < slots); i++)
```

Comment:
Using `int` instead of `char` seems natural since `slots` is `int` type. And `(` and `)` are redundant.